src/gh/components/DF_csv_exporter/code.py

```diff
@@ -1,11 +1,45 @@
 #! python3
 
+import System
+import csv
+import os
+import typing
 
 from ghpythonlib.componentbase import executingcomponent as component
+import Grasshopper as gh
 
-from diffCheck.df_error_estimation import DFInvalidData
-import csv
-import os
+from diffCheck.df_error_estimation import DFInvalidData, DFVizResults
+
+
+def add_bool_toggle(self,
+    nickname: str,
+    indx: int,
+    X_param_coord: float,
+    Y_param_coord: float,
+    X_offset: int=87
+    ) -> None:
+    """
+        Adds a boolean toggle to the component input
+
+        :param nickname: the nickname of the value list
+        :param indx: the index of the input parameter
+        :param X_param_coord: the x coordinate of the input parameter
+        :param Y_param_coord: the y coordinate of the input parameter
+        :param X_offset: the offset of the value list from the input parameter
+    """
+    param = ghenv.Component.Params.Input[indx]  # noqa: F821
+    if param.SourceCount == 0:
+        toggle = gh.Kernel.Special.GH_BooleanToggle()
+        toggle.NickName = nickname
+        toggle.Description = "Toggle the value to use with DFVizSettings"
+        toggle.CreateAttributes()
+        toggle.Attributes.Pivot = System.Drawing.PointF(
+            X_param_coord - (toggle.Attributes.Bounds.Width) - X_offset,
+            Y_param_coord - (toggle.Attributes.Bounds.Height / 2 + 0.1)
+            )
+        toggle.Attributes.ExpireLayout()
+        gh.Instances.ActiveCanvas.Document.AddObject(toggle, False)
+        ghenv.Component.Params.Input[indx].AddSource(toggle)  # noqa: F821
 
 
 class DFCsvExporter(component):
```

Comment:
Hi ! I didn't see a check on the number of character in a csv cell, to make sure the csv file is readable. Is it something that should be implemented later ? I can do it, no problem, but it's just to make sure we don't miss it. I guess we could simply add a warning to the component if the length of any of the "distances" of the row dictionnary is over the limit (for example 6553 distances if we keep 4 digits, because the [limit is to 32767 in excel](https://stackoverflow.com/questions/9971542/field-size-limitation-of-csv-file).

Reply:
there is none because right now, the way it is we will never export a cell with a large number of chars. That was the only case of the "distances" cell. With this refactoring we are exporting the distances separate, each distance in a different row. So no need for a check.

Reply:
Also we do not have CSV reader yet.